### ISSUE TITLE
Drives not showing up on home screen

### ISSUE THREAD

**Issue by KibSquib48** [files-community/Files#3231]
**Describe the bug**
Drives do not appear on home screen until opening one of the library folders, and sometimes not even that works.

**To Reproduce**
1. Open files
2. Go to click on the C: Drive
3. wait where did it go

**Expected behavior**
The buttons for the C: drive and all other drives should be there

**Screenshots**
![image](https://user-images.githubusercontent.com/33189136/105526413-5f6e5d80-5cb0-11eb-8672-1239934b84db.png)

**Desktop (please complete the following information):**
 - OS Version: Windows 10 20H2 19042.782
 - App version: 1.1.5.0

**Additional context**
Terminating, Resetting, and Repairing from the Settings app does not fix it, and neither does a simple close and reopen.

**Log file**
[debug.txt](https://github.com/files-community/Files/files/5857617/debug.txt)
[debug_fulltrust.txt](https://github.com/files-community/Files/files/5857619/debug_fulltrust.txt)

**Comment by gave92** [comment]
@KibSquib48 does it work when you type a drive path e.g. "C:\\" in the navigationbar?

**Comment by SimonCDalzell** [comment]
Typing in the path of one of the non-showing drives works and contents is displayed correctly.

**Comment by gave92** [comment]
@SimonCDalzell do you experience the same? Is all your drives missing or just some? Could you post the log files a screenshot of the sidebar?
Are the drive missing from the sidebar only or from the home page too?

**Comment by KibSquib48** [comment]
> @KibSquib48 does it work when you type a drive path e.g. "C:\" in the navigationbar?
> Did the missing drive by chance reappear after installing OneDrive?

Installing OneDrive fixed it, and typing C:\ worked too

**Comment by SimonCDalzell** [comment]
> @SimonCDalzell do you experience the same?
> 
> * are all your drives missing or just some?
> * could you post the log files and a screenshot of the sidebar?
> * are the drive missing from the sidebar only or from the home page too?

Just some. I believe the two drives that appear correctly have AHCI Hotswap enabled in the bios, while the ones that do not show have it disabled. I can test this in a sec. 
I've attached the logs and a screenshot
The drives are missing from the sidebar and from the home page.

Probably worth mentioning that I'm on Windows 1809.

![Capture](https://user-images.githubusercontent.com/19229302/105580129-3ffc3100-5d93-11eb-9957-00e14bb93121.JPG)
[debug.txt](https://github.com/files-community/Files/files/5860295/debug.txt)
[debug_fulltrust.txt](https://github.com/files-community/Files/files/5860296/debug_fulltrust.txt)

**Comment by gave92** [comment]
@SimonCDalzell could you verify that:
In Windows Settings > Privacy > File system permission is turned on for Files

**Comment by SimonCDalzell** [comment]
> @SimonCDalzell could you verify that:
> In Windows Settings > Privacy > File system permission is turned on for Files

That solved it. I had not received the prompt to enable file access. I then unplugged the drives U and V, and next time I launched files was prompted to enable file system permissions.

**Comment by gave92** [comment]
@SimonCDalzell thanks for confirmation, the prompt is only shown if Files is not able to access any fixed drive. What kind of drive where U and V? Usb HDDs?

**Comment by SimonCDalzell** [comment]
> @SimonCDalzell thanks for confirmation, the prompt is only shown if Files is not able to access any fixed drive. What kind of drive where U and V? Usb HDDs?

All are internal drives connected to motherboard SATA ports. I unplugged U and V for an unrelated reason and happened to notice the prompt for file access. As far as I remember of my configuration the only difference between the drives is the AHCI Hotswap mode. Happy to investigate more thoroughly though.

**Comment by XPoppyX** [comment]
In my case it takes a long time to load drives.


![loading](https://user-images.githubusercontent.com/61259627/105616008-6bdadf00-5db2-11eb-9ff9-080715f45063.gif)

**Comment by gave92** [comment]
@XPoppyX thanks for the report. That's unfortunately expected after #3177. Working on it :)

**Comment by AndyMDoyle** [comment]
@XPoppyX, if you are in a position to do so, please try out the commits from PR #3257.
This PR should allow local drives to appear almost instantly, while allowing cloud drives to appear a little later.

Or if you disconnect drive D (which appears to be a network or mapped drive), does that populate your drives list quicker?

**Comment by XPoppyX** [comment]
Start faster now (without disconnecting the network drive)

**Comment by AndyMDoyle** [comment]
@XPoppyX  thanks for trying that. Would you say it is acceptable performance with your setup or do you think there is room for improvement?

**Comment by XPoppyX** [comment]
Well, any speed improvement is welcome, because it is still slower than the native file explorer.

2 seconds vs. less than 400 ms

I guess because it's UWP it complicates things more

**Comment by AndyMDoyle** [comment]
UWP is a strange beast. Unlike WinForms which had minimal difference between debug and release modes, UWP apps are so much slower in debug mode and it makes it hard to troubleshoot issues.

I agree Files is slower at launching that file explorer, and I have been personally trying to improve this. Still a few ideas in the works to propose but I don't know if it would be in the direction everyone wants to go.

**Comment by scottkuhl** [comment]
I have an issue where instead of my OneDrive showing up, sometimes it shows the C Drive twice.

![image](https://user-images.githubusercontent.com/4465616/106089124-ef673980-60ec-11eb-8303-bdab3e80b235.png)

**Comment by gave92** [comment]
@scottkuhl but if you open a new tab, the new tab sidebar shows the correct items, right?

**Comment by AndyMDoyle** [comment]
@gave92 @scottkuhl I have had something similar occur while finishing the sync'd SharePoint folder integration. The last entry added to the side bar is intermittently duplicated, or even stranger, causing the last-1 item to be rendered as a duplicate of the last item.

Locally I have code logging all of the items discovered, logging the order they are being added to the sidebar, and then dump out the current items from `MainPage.SideBarItems`. What is confusing me is all of this logging shows exactly what I expect - everything is correct, in the right order, and there is no duplication. But the `NavigationView` is not displaying them as they appear in `MainPage.SideBarItems`.

I am digging through the microsoft-ui-xaml repo now for related issues to see if this is a known problem.

**Comment by gave92** [comment]
@AndyMDoyle yeah that's what I meant by "visual duplicates", looks like a bug in NavigationView. Calling a Collection.Reset on the sidebar list usually fixes the visualization.

**Comment by scottkuhl** [comment]
> @scottkuhl but if you open a new tab, the new tab sidebar shows the correct items, right?

@gave92  That is correct.

**Comment by yaichenbaum** [comment]
@KibSquib48 This issue should be resolved with #3350.
